### PR TITLE
List Py dependencies in a requirements.txt file

### PR DESCRIPTION
**Pull request by nilsvu**
## Proposed changes

Makes it easier for users to install the dependencies. Hopefully reduces the number of people using the wrong yapf version as well.

### Code review checklist

- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).
- [ ] The PR lists upgrade instructions and is labeled `bugfix` or
  `new feature` if appropriate.

### CODE REVIEW

File: docs/Installation/Installation.md [hunk 1]
```diff
@@ -78,35 +78,31 @@ all of these dependencies.
...
   Building with shared library support is also recommended.
 * [Python](https://www.python.org/) 3.7 or later.
-* [NumPy](http://www.numpy.org/) 1.10 or later
-* [SciPy](https://www.scipy.org)
-* [matplotlib](https://matplotlib.org/)
-* [PyYAML](https://pyyaml.org/)
-* [h5py](https://www.h5py.org)
+* Python dependencies listed in `support/Python/requirements.txt`.
+  Install with `pip3 install -r support/Python/requirements.txt`.
+  Make sure you are working in a [Python venv])(https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment)
```
Comment (wthrowe) [inline docs/Installation/Installation.md]: Extra `)` breaking the link.

File: docs/Installation/Installation.md [hunk 2]
```diff
@@ -78,35 +78,31 @@ all of these dependencies.
...
   generate documentation
-* [Python](https://www.python.org/) with
-  [BeautifulSoup4](https://pypi.org/project/beautifulsoup4/),
-  [Pybtex](https://pybtex.org),
-  [nbconvert](https://nbconvert.readthedocs.io/)
-  — for documentation pre- and post-processing
+* Python dev dependencies listed in `support/Python/dev_requirements.txt`
+  — for documentation pre- and post-processing, formatting code, etc.
+  Install with `pip3 install -r support/Python/dev_requirements.txt`.
+  Make sure you are working in a [Python venv])(https://packaging.python.org/en/latest/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment)
```
Comment (wthrowe) [inline docs/Installation/Installation.md]: Extra paren again.

**Review by wthrowe — CHANGES_REQUESTED** [review]
Would it be possible to get the lists included in the documentation as snippets or something?  It's kind of nice only having to look in one place to see all the dependencies.  Not a big deal if it is hard.

**Comment by nilsvu** [comment]
Good idea @wthrowe, added a fixup. Also added a commit using the requirements.txt in the Dockerfile.

**Comment by wthrowe** [comment]
I didn't know about the "details" thing.  That looks nice.  Squash.

**Review by wthrowe — APPROVED** [review]

**Review by nilsdeppe — APPROVED** [review]

**Pull request merged**